compose/ui/ui-text/src/desktopMain/kotlin/androidx/compose/ui/text/platform/AwtFontInterop.kt

```diff
@@ -0,0 +1,161 @@
+ * weight of the [Font], in addition to the _actual_ font family name. This
+ * can cause issues when trying to look up the corresponding Compose font
+ * families. Besides that, it will result in duplicate entries when listing
+ * font families.
+ *
+ * **Note:** this will return `null` when running on Java VMs other
+ * than the JetBrains Runtime, and/or if the current module cannot
+ * access the `sun.font` module. To ensure the `sun.font` module is
+ * accessible, you should pass this argument to the JVM: `--add-opens
+ * java.desktop/sun.font=ALL-UNNAMED`.
+ */
+@ExperimentalTextApi
+fun Font.fontFamilyNameOrNull(): String? = AwtFontUtils.getPreferredFontFamilyName(this)
+
+/**
+ * Returns the path of the file that a [Font] is loaded from; it will be
+ * `null` if the font is not backed by a [sun.font.FileFont].
+ *
+ * **Note:** this will return `null` when the current module cannot
+ * access the `sun.font` module. To ensure the `sun.font` module is
+ * accessible, you should pass this argument to the JVM: `--add-opens
+ * java.desktop/sun.font=ALL-UNNAMED`.
+ */
+@ExperimentalTextApi
+fun Font.physicalFontFileOrNull(): String? = AwtFontUtils.fontFileNameOrNull(this)
+
+/**
+ * Try to resolve a font family name, that could be a logical font face
+ * (e.g., [Font.DIALOG]), to the actual physical font family it is an alias
+ * for.
+ *
+ * **Note:** this will return `null` when running on Java VMs other
+ * than the JetBrains Runtime, and if the current module cannot
+ * access the `sun.font` module. To ensure the `sun.font` module is
+ * accessible, you should pass this argument to the JVM: `--add-opens
+ * java.desktop/sun.font=ALL-UNNAMED`.
+ *
+ * @return The resolved physical font name, or `null` if it can't be
+ *     resolved (either it's unknown, or [isAbleToResolveFontProperties] is
+ *     false)
+ */
+@ExperimentalTextApi
+fun Font.resolveFontFamilyNameOrNull(): String? =
+    AwtFontUtils.resolvePhysicalFontFamilyNameOrNull(fontFamilyNameOrNull() ?: family, style)
+
+/**
+ * Load a [FontFamily] from a font family name. The Compose runtime is able
+ * to load font families available in the system, and ones embedded in the
+ * JetBrains Runtime (see note below).
+ *
+ * If a family is not known to Compose, the lookup will return a fallback
+ * font family.
+ *
+ * If you're trying to use an AWT [Font] in Compose, use the
+ * [Font.asFontFamily] function instead, which will take care of some
+ * AWT-specific quirks, too.
+ *
+ * **Note:** when running on Java VMs other than the JetBrains Runtime,
+ * only system fonts are available. Embedded JVM fonts on other runtimes
+ * are not supported.
+ *
+ * @param familyName The name of the font family to load
+ * @param onlySystemFonts If true, embedded JBR fonts are ignored
+ * @see Font.asFontFamily
+ */
+@ExperimentalTextApi
+@Stable
+fun FontFamily(familyName: String, onlySystemFonts: Boolean = false): FontFamily {
```

Comment:
This function isn't AWT related in its signature. It can conflict with other potential function that have the same signature, but its implementation uses Skia API to load a FontFamily by a name.

Can we make it internal?